src/modules/Update.rb

```diff
@@ -773,47 +773,96 @@ def installed_product
+    #
     # @example to store repos file and credentials directory
-    #   Update.create_backup("0100-repos", ["/etc/zypp/repos.d/*", "/etc/zypp/credentials"])
+    #   Yast::Update.create_backup("0100-repos", ["/etc/zypp/repos.d/*", "/etc/zypp/credentials"])
     def create_backup(name, paths)
-      log.info "Creating tarball for #{name} including #{paths}"
       mounted_root = Installation.destdir
 
+      # Copy the os-release file (bsc#1097297)
+      log.info "Copying the os-release file"
+      copy_os_release
+
+      log.info "Creating the tarball for #{name} including #{paths}"
       tarball_path = File.join(BACKUP_DIR, "#{name}.tar.gz")
       root_tarball_path = File.join(mounted_root, tarball_path)
       create_tarball(root_tarball_path, mounted_root, paths)
 
+      log.info "Creating the restore script for #{name}"
       script_path = File.join(mounted_root, BACKUP_DIR, "restore-#{name}.sh")
       create_restore_script(script_path, tarball_path, paths)
     end
 
-    # clean backup content. Usefull to clean up all content before creating new backup
+    # Removes the backup content
+    #
+    # Usefull to clean up **ALL** content in BACKUP_DIR before creating a new backup.
     def clean_backup
       log.info "Cleaning backup dir"
-      mounted_root = Installation.destdir
-      ::FileUtils.rm_r(File.join(mounted_root, BACKUP_DIR),
-        :force => true, :secure => true)
+      ::FileUtils.rm_r(File.join(Installation.destdir, BACKUP_DIR), :force => true, :secure => true)
     end
 
-    # restores backup
+    # Restores the available backup(s)
+    #
+    # If the VERSION_ID in the backup os-release file matches with the value stored in
+    # /etc/os-release (see bsc#1097297), the available backup scripts (restores-*.sh) will be
+    # executed **in the expected order** (see bsc#1089643).
+    #
+    # @see #restore_backup?
     def restore_backup
-      log.info "Restoring backup"
-      mounted_root = Installation.destdir
-      script_glob = File.join(mounted_root, BACKUP_DIR,"restore-*.sh")
-      # sort the scripts to execute them in the expected order
-      ::Dir.glob(script_glob).sort.each do |path|
-        cmd = "sh #{path} #{File.join("/", mounted_root)}"
-        res = SCR.Execute(path(".target.bash_output"), cmd)
-        log.info "Restoring with script #{cmd} result: #{res}"
+      if restore_backup?
+        log.info "Restoring backup"
+
+        mounted_root = Installation.destdir
+        available_restore_scripts = ::Dir.glob(File.join(mounted_root, BACKUP_DIR, "restore-*.sh"))
+
+        available_restore_scripts.sort.each do |path|
+          cmd = "sh #{path} #{File.join("/", mounted_root)}"
+          res = SCR.Execute(path(".target.bash_output"), cmd)
+
+          log.info "Restoring with script #{cmd} result: #{res}"
+        end
+      else
+        log.error "Backup was not restored because its version info does not match"
```

Comment:
I think it would be nice to log some more details, like what was found and what was expected. That will make debugging easier. Because that data is not available here I'm fine with logging it in the `restore_backup?` method.